### PR TITLE
monitoring-dam-generator minor refactoring + fixed exception handling + now it returns application level monitoring rules and the already updated ADP.

### PR DESCRIPTION
**Pull request by MicheleGuerriero**
This PR adds the following fixes/features to the monitoring-dam-generator-core:

- better exception handling during the parsing of the ADP
- the monitoring-dam-generator-core now returns directly application level information, i.e. the application entire set of monitoring rules (instead of returning module specific sets of rules) and the ADP already modified with the data collectors nodeTemplates (according to the current definition in the DAM for data collectors)
- the DAMGenerator has been modified in order to be compliant with the new way the monitoring-dam-generator-core works. 
- minor code and tests refactoring of the monitoring-dam-generator-core

@szenzaro since you are the ojne that will use the components, can you review this code?

### CODE REVIEW

File: monitor/monitoring-dam-generator-core/resources/currentAtosAdpFromOptimizer.yml [hunk 1]
```diff
@@ -29,8 +30,8 @@ topology_template:
           - {greater_or_equal: '5.0'}
           - {less_or_equal: '5.6'}
       requirements:
-      - {host: Amazon_EC2_i2_xlarge_us_west_2, instancesPOC: 1}
-    Amazon_EC2_i2_xlarge_us_west_1:
+      - {host: AmazonEC2i2xlargeuswest2, instancesPOC: 1}
+    AmazonEC2i2xlargeuswest1:
```
Comment (szenzaro) [inline monitor/monitoring-dam-generator-core/resources/currentAtosAdpFromOptimizer.yml]: Does the name format affect the output result?

File: byon/files/temporal-catalog.txt [hunk 2]
```diff
@@ -0,0 +1,368 @@
+brooklyn.catalog:
```
Comment (szenzaro) [inline byon/files/temporal-catalog.txt]: Probably this file should not part of the PR. wdyt @andreaturli ?

File: monitor/monitoring-dam-generator-core/resources/currentAtosAdpFromOptimizer.yml~ [hunk 3]
```diff
@@ -0,0 +1,119 @@
+tosca_definitions_version: tosca_simple_yaml_1_0_0_wd03
```
Comment (szenzaro) [inline monitor/monitoring-dam-generator-core/resources/currentAtosAdpFromOptimizer.yml~]: Probably this file is a backup file of your editor. May we remove it from the PR?

File: monitor/monitoring-dam-generator-core/src/main/java/eu/seaclouds/monitor/monitoringdamgenerator/MonitoringDamGenerator.java [hunk 4]
```diff
@@ -15,80 +14,56 @@
...
-    
+
     private URL monitoringManagerUrl;
-    
-    
-    private static final String NURO_DC_ID="nuroDc";
-    private static final String JAVA_APP_DC_ID="javaAppDc";
-    private static final String MODACLOUDS_DC_ID="modacloudsDc";
 
     public MonitoringDamGenerator(String monitorIp, String monitorPort) {
```
Comment (szenzaro) [inline monitor/monitoring-dam-generator-core/src/main/java/eu/seaclouds/monitor/monitoringdamgenerator/MonitoringDamGenerator.java]: Using URL as parameter should be enough.

File: monitor/monitoring-dam-generator-core/src/main/java/eu/seaclouds/monitor/monitoringdamgenerator/adpparsing/Module.java [hunk 5]
```diff
@@ -67,52 +71,16 @@ public void setAvailability(double availability) {
...
-
+    
     @Override
     public String toString() {
         String info = "";
-        info += "ModuleName= " + this.moduleName + " RespTime= " + this.respTime + " Availability= "
-                + this.availability + " Cost= " + this.getCostMonth()
-                + " Workload= " + this.getWorkload() +"Host= " + this.host.getHostName()
-                + " DeploymentType= " + this.host.getDeploymentType() + " isJavaApp= "+this.isJavaApp;
+        info += "ModuleName= " + this.moduleName
```
Comment (szenzaro) [inline monitor/monitoring-dam-generator-core/src/main/java/eu/seaclouds/monitor/monitoringdamgenerator/adpparsing/Module.java]: Use StringBuilder for multiple string concatenation.

File: planner/core/src/main/java/eu/seaclouds/platform/planner/core/DamGenerator.java [hunk 6]
```diff
@@ -1,7 +1,7 @@
 package eu.seaclouds.platform.planner.core;
```
Comment (szenzaro) [inline planner/core/src/main/java/eu/seaclouds/platform/planner/core/DamGenerator.java]: Could you please rebase the branch maintaining this changes history?

File: byon/files/temporal-catalog.txt (hunk 2 above)
Comment (MicheleGuerriero) [inline byon/files/temporal-catalog.txt]: You are right, it was there for testing other things, I'm sorry. I'm going to remove it.

File: monitor/monitoring-dam-generator-core/resources/currentAtosAdpFromOptimizer.yml~ (hunk 3 above)
Comment (MicheleGuerriero) [inline monitor/monitoring-dam-generator-core/resources/currentAtosAdpFromOptimizer.yml~]: Right, done.

File: monitor/monitoring-dam-generator-core/src/main/java/eu/seaclouds/monitor/monitoringdamgenerator/MonitoringDamGenerator.java (hunk 4 above)
Comment (MicheleGuerriero) [inline monitor/monitoring-dam-generator-core/src/main/java/eu/seaclouds/monitor/monitoringdamgenerator/MonitoringDamGenerator.java]: Ok, done. Now it receives just and URL parameter.

File: monitor/monitoring-dam-generator-core/src/main/java/eu/seaclouds/monitor/monitoringdamgenerator/adpparsing/Module.java (hunk 5 above)
Comment (MicheleGuerriero) [inline monitor/monitoring-dam-generator-core/src/main/java/eu/seaclouds/monitor/monitoringdamgenerator/adpparsing/Module.java]: Right, done.

File: monitor/monitoring-dam-generator-core/resources/currentAtosAdpFromOptimizer.yml (hunk 1 above)
Comment (MicheleGuerriero) [inline monitor/monitoring-dam-generator-core/resources/currentAtosAdpFromOptimizer.yml]: Sorry this is something temporary while I was testing. Now I fixed it.

**Comment by szenzaro** [comment]
+1 for me. Can we merge?
